gcloud/credentials.py

```diff
@@ -177,12 +174,14 @@ def _get_pem_key(credentials):
     :raises: `TypeError` if `credentials` is the wrong type.
              `EnvironmentError` if `crypto` did not import successfully.
     """
-    if isinstance(credentials, client.SignedJwtAssertionCredentials):
-        # Take our PKCS12 (.p12) text and convert to PEM text.
-        pem_text = crypt.pkcs12_key_as_pem(credentials.private_key,
-                                           credentials.private_key_password)
-    elif isinstance(credentials, service_account._ServiceAccountCredentials):
-        pem_text = credentials._private_key_pkcs8_text
+    if isinstance(credentials, ServiceAccountCredentials):
+        if credentials._private_key_pkcs12 is not None:
+            # Take our PKCS12 (.p12) text and convert to PEM text.
+            pem_text = crypt.pkcs12_key_as_pem(
+                credentials._private_key_pkcs12,
```

Comment:
As of right now, they are the only chunks that signal the difference. However, I want to (soon) upstream all of this and just use `credentials.sign_blob`